### PR TITLE
Remove view and add withView

### PR DESCRIPTION
**Pull request by janjelinek**
Absolutely not ready for merge, it's just idea for issue #8 when view will be removed could be helpful have option, how to add view to same place where rest of config of Glued module is.

### CODE REVIEW

File: docs/index.html [hunk 1]
```diff
@@ -9247,24 +9254,28 @@
...
+				}),
+			init: _user$project$Bubbling_Counter$init(_user$project$Bubbling_Main$Even),
+			update: _user$project$Bubbling_Counter$update(_user$project$Bubbling_Main$Even),
+			subscriptions: function (_p0) {
+				return _elm_lang$core$Platform_Sub$none;
+			}
+		}));
 var _user$project$Bubbling_Main$subscriptions = A2(
 	_user$project$Glue$subscriptions,
 	_user$project$Bubbling_Main$counter,
```
Comment (turboMaCk) [inline docs/index.html:9281]: this should be rebuilt only when release is made.

File: src/Glue.elm [hunk 2]
```diff
@@ -130,29 +130,43 @@ poly :
...
+```
+withView :
+    (model -> Html msg)
+    -> Glue model subModel msg subMsg
+    -> Glue model subModel msg subMsg
+```
+-}
+withView : (model -> Html msg) -> Glue model subModel msg subMsg -> Glue model subModel msg subMsg
+withView view (Glue data) =
+    Glue { data | view = view }
```
Comment (turboMaCk) [inline src/Glue.elm:167]: This implementation won't be possible. There won't be `view` in `Glue`'s type record anymore - you won't be able to store this function anywhere.

**Comment by turboMaCk** [comment]
Thanks for PR. Anyway something like this won't be possible once `view` will be removed from `Glue`'s internal record.

Anyway There will be function `Glue.view` just will work differently. It will expect one extra argument (`subModel -> Html msg`). This is example usage of future implementation:

```elm
view : Model -> Html Msg
view model =
    Glue.view subModule SubModule.view model
```

advantage will be that this will make possible things like:

```elm
view : Route -> Model -> Html Msg
view route model =
    Glue.view subModule (SubModule.view route) model
```

On the other hand it will have some disadvantages as well. It will no longer be possible to reflect `poly` or `simple` mapping in this function so if child `Html` will holds its own `Msg`, `map` will be required like:

```elm
view : Model -> Html Msg
view model =
    Glue.view subModule (Html.Map ChildMsg << SubModule.view) model
```

File: src/Glue.elm [hunk 3]
```diff
@@ -130,29 +130,43 @@ poly :
...
         , init = init
         , update =
             \subMsg model ->
                 get model
                     |> update subMsg
-        , view =
-            \model ->
-                get model
-                    |> view
+        , view = \model -> text ""
```
Comment (turboMaCk) [inline src/Glue.elm:144]: I think we should remove this from type itself. Mocking something implicitly seems to be confusing to me.

**Comment by janjelinek** [comment]
I like this 
```
view : Route -> Model -> Html Msg
view route model =
    Glue.view subModule (SubModule.view route) model
```
it make sense, my solution was expecting keep `view` in Glue record, but you're right with this https://github.com/turboMaCk/glue/pull/9#discussion_r117314177

File: docs/index.html (hunk 1 above)
Comment (janjelinek) [reply docs/index.html:9281]: I was testing examples and this was builded during it, I did not pay attention, it wasn't important for my question :) But good to know 👍

**Comment by turboMaCk** [comment]
Btw you can check updates in issue. I had some ideas for improving `view` support there https://github.com/turboMaCk/glue/issues/8#issuecomment-302511645

File: docs/index.html (hunk 1 above)
Comment (turboMaCk) [reply docs/index.html:9281]: no worries;)

**Comment by turboMaCk** [comment]
solved in https://github.com/turboMaCk/glue/releases/tag/2.0.0

**Pull request closed without merging**